torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2952,9 +2990,9 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::pointToPoint(
   }
 
   // Used many times below, so we stash the unordered_map lookup
-  auto ncclStream = ncclStreams_.at(key);
+  auto ncclStream = ncclStreams_.at(p2pKey);
   // First let NCCL streams wait for input tensors allocation streams
-  syncStream(device, ncclEvents_[key], ncclStream);
+  syncStream(device, ncclEvents_[p2pKey], ncclStream);
```

Comment:
In the old logic this is conditionally the p2pKey or the devKey -- is it intentional to always use the p2p key now?

Reply:
yes- it is intentional to always use the p2pkey for the stream, based on the wrong assumption that using the same comm but different stream would allow overlap between p2p ops involving different peers.

but i suspect i missed something- i probably should have kept this as devKey for batched-p2p ops and only made this p2pkey for true p2p ops.